server/src/YmlToBdl.g4

```diff
@@ -118,50 +119,41 @@ ENUM : 'enum';
 //Vérifier adéquation avec les terminaux définis par Olivier
 ID : LETTER ALPHANUM* ;
 
-//We store the whitespace in hidden channel to take back the ones around comments
-
 fragment MULTILINE_COMMENT_START: '/*';
 fragment MULTILINE_COMMENT_END: '*/';
 
-LINE_COMMENT : '//' ~[\n\r]* ;
-MULTILINE_COMMENT: MULTILINE_COMMENT_START .*? MULTILINE_COMMENT_END;
+LINE_COMMENT : '//' ~[\n\r]*  -> channel(HIDDEN);
+MULTILINE_COMMENT: MULTILINE_COMMENT_START .*? MULTILINE_COMMENT_END -> channel(HIDDEN);
 GRANULE: OPEN_GRANULE (.)*? CLOSE_GRANULE EOF?;
 PREPROCESSING: '@' ~[\n\r]* ;
 
+//Parser rules
 expressionMarker : DOT
                  | MULTIVALUED_EXPRESSION;
-commentElement : WS? (LINE_COMMENT | MULTILINE_COMMENT) WS? ;
-
 preprocessingElement : WS? PREPROCESSING WS?;
 
-comment :  commentElement  ;
-
-commentBlock: (comment | preprocessingElement)+;
-
-
 freeText : .+? (~EOF)*? ;
 
-//Parser rules
 
+// Colons aren't interpreted by YE in a particular way.
 ymlId : ID (COLON? COLON ID)*;
-//Les : ne sont pas interprétés par l'engine de manière particulière
 
+// Base rule when parsing a file. Basicaly, this describes all valid YML files.
 kaoFile : entities=ymlEntity* finalFreeText=freeText? EOF;
-yclassFile : classDeclaration restFreeText=freeText? EOF?;
 
 ymlEntity: classDeclaration
             | staticDeclaration
             | complete
             | methodDeclaration SEMICOLON
             | yenum;
 
-yenum: ENUM yid=ymlId OPEN_BRACE (enumElement (beforeCommaComment=commentBlock? COMMA enumElement)*)+ beforeCloseBraceComment=commentBlock? CLOSE_BRACE fields=field* SEMICOLON;
+yenum: ENUM yid=ymlId OPEN_BRACE (enumElement ( COMMA enumElement)*)+ CLOSE_BRACE fields=field* SEMICOLON;
 enumElement : yid=ymlId fields=field* ;
 
 //Field and options
-classDeclaration : beforeInterfaceComment=commentBlock? classDeclarationIntro classOption=field* memberDeclaration* methodDeclaration* beforeClassPropertyComment=commentBlock? classPropertiesBlock? endInterfaceComment=commentBlock? SEMICOLON intermediateFreeText=freeText*? classImplementation SEMICOLON EOF?;
+classDeclaration :  classDeclarationIntro field* (memberDeclaration | methodDeclaration)*  classPropertiesBlock?  SEMICOLON intermediateFreeText=freeText*? classImplementation SEMICOLON EOF?;

```

Comment:
Thanks to ` (memberDeclaration | methodDeclaration)* ` we can now mix both types without problems.